Lib/asyncio/base_events.py

```diff
@@ -867,6 +925,63 @@ def _check_sendfile_params(self, sock, file, offset, count):
 
         return transport, protocol
 
+    async def sendfile(self, transport, file, offset=0, count=None,
+                       *, fallback=True):
+        """Send a file through a transport.
+
+        Return amount of sent bytes.
+        """
+        if transport.is_closing():
+            raise RuntimeError("Transport is closing")
+        mode = getattr(transport, '_sendfile_compatible',
+                       constants._SendfileMode.UNSUPPORTED)
+        if mode is constants._SendfileMode.UNSUPPORTED:
+            raise RuntimeError(
+                f"sendfile is not supported for transport {transport!r}")
+        if mode is constants._SendfileMode.TRY_NATIVE:
+            try:
+                return await self._sendfile_native(transport, file,
+                                                   offset, count)
+            except events.SendfileNotAvailableError as exc:
+                if not fallback:
+                    raise
+        # the mode is FALLBACK or fallback is True
+        assert (mode is constants._SendfileMode.FALLBACK or mode is
+                constants._SendfileMode.TRY_NATIVE and fallback), mode
```

Comment:
I know I asked you to add this assert.. Let's remove it :) I think it's too big and it actually distracts the reader more than it helps them.